src/server/routes/agent.rs

```diff
@@ -166,6 +166,10 @@ fn endpoint_record_progress(
         ex.name, auth.name,
     );
 
+    data.metric
+        .with_label_values(&[&auth.name, &ex.name])
+        .inc_by(result.data.results.len() as i64);
```

Comment:
When you create the new struct it would be nice for this to be a method call, to avoid exposing the metric directly.